packages/react-vapor/src/components/unsavedChangeModal/UnsavedChangeModalProvider.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import {Button} from '../button/Button';
+import {ModalComposite} from '../modal/ModalComposite';
+
+const defaultModalTitle = 'Unsaved Changes';
+const defaultModalClasses = ['mod-prompt', 'mod-fade-in-scale'];
+const defaultModalDescription =
+    'Are you sure you want to leave this page without saving? All unsaved changes will be lost.';
+const defaultConfirmButtonText = 'Exit without applying changes';
+
+export interface IUnsavedChangesModalProviderProps {
+    isDirty: boolean;
+    modalTitle?: string;
+    className?: string[];
+    children: (props: {promptBefore: (callbackOnExit: () => any) => boolean}) => React.ReactNode;
```

Comment:
I think `callbackOnExit` could be changed to `callbackOnDiscard`, I feel it would clear the intent a little

Reply:
:+1:  Clearer is better